sdk/textanalytics/cognitiveservices-textanalytics/README.md

```diff
@@ -21,30 +28,89 @@ Use the client library to:
+
 const client = new TextAnalyticsClient(
   "<endpoint>",
   new CognitiveServicesCredentials("<service key>")
 );
 ```
 
+#### Using an Azure Active Directory Credential
+
+Client subscription key authentication is used in most of the examples, but you can also authenticate with Azure Active Directory using the [Azure Identity library][azure_identity]. To use the [DefaultAzureCredential][defaultazurecredential] provider shown below,
+or other credential providers provided with the Azure SDK, please install the `@azure/identity` package:
+
+```bash
+npm install --save @azure/identity
+```
+
+You will also need to [register a new AAD application][register_aad_app] and grant access to Text Analytics by assigning the `"Cognitive Services User"` role to your service principal.
+
+Set the values of the client ID, tenant ID, and client secret of the AAD application as environment variables: AZURE_CLIENT_ID, AZURE_TENANT_ID, AZURE_CLIENT_SECRET.
+
+```js
+const { TextAnalyticsClient } = require("@azure/cognitiveservices-textanalytics");
+const { DefaultAzureCredential } = require("@azure/identity");
+
+const client = new TextAnalyticsClient("<endpoint>", new DefaultAzureCredential());
+```
+
 ## Key concepts
 
-Content still being written.
+### TextAnalyticsClient
+
+`TextAnalyticsClient` is the primary interface for developers using the Text Analytics client library. It provides asynchronous methods to access a specific use of Text Analytics, such as language detection or key phrase extraction.
+
+### Text Input
+
+A **text input**, sometimes called a **document**, is a single unit of input to be analyzed by the predictive models in the Text Analytics service. Operations on `TextAnalyticsClient` take a collection of inputs to be analyzed as a batch.
+
+### Operation Result
+
+An operation result, such as `AnalyzeSentimentResult`, is the result of a Text Analytics operation, containing a prediction or predictions about a single text input. An operation's result type also may optionally include information about the input document and how it was processed.
+
+### Operation Result Collection
+
+An operation result collection, such as `AnalyzeSentimentResultCollection`, is a collection of operation results, where each corresponds to one of the text inputs provided in the input batch. A text input and its result will have the same index the input and result collections. An operation result collection may optionally include information about the input batch and how it was processed.
+
+### Operation Overloads
+
+For each supported operation, `TextAnalyticsClient` provides method overloads to take a batch of text inputs as strings or a batch of `LanguageInput` or `MultiLanguageInput` objects, depending on the operation. Overloads that take an object batches allows callers to give each document a unique ID, or indicate extra metadata such as the langauge the document is written in or the country of origin.
```

Comment:
If you're still using LanguageInput and MultiLanguageInput, would you be able to bring those into alignment with the other languages' types names?

Reply:
The proposed naming would be `DetectLanguageInput` and `TextDocumentInput`, right? These come from the swagger, so I'd have to patch that to do the rename. 

@bterlson any feelings here?

Reply:
No real feelings, but happy to do the alignment.

Reply:
Alignment done!